pennylane/ops/op_math/controlled_ops.py

```diff
@@ -0,0 +1,124 @@
+    * ``target_wires``: the wires the unitary matrix will be applied to
+    * ``active_wires``: Wires modified by the operator. This is the control wires followed by the target wires.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (2,)
+    * Gradient recipe: None
+
+    Args:
+        U (Union[array[complex], QubitUnitary]): square unitary matrix or a QubitUnitary operation
+        control_wires (Union[Wires, Sequence[int], or int]): the control wire(s)
+        wires (Union[Wires, Sequence[int], or int]): the wire(s) the unitary acts on (optional if U is provided as a QubitUnitary)
+        control_values (List[int, bool]): a list providing the state of the control qubits to control on (default is the all 1s state)
+        unitary_check (bool): whether to check whether an array U is unitary when creating the operator (default False)
+
+    **Example**
+
+    The following shows how a single-qubit unitary can be applied to wire ``2`` with control on
+    both wires ``0`` and ``1``:
+
+    >>> U = np.array([[ 0.94877869,  0.31594146], [-0.31594146,  0.94877869]])
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1], wires=2)
+
+    Alternatively, the same operator can be constructed with a QubitUnitary:
+
+    >>> base = qml.QubitUnitary(U, wires=2)
+    >>> qml.ControlledQubitUnitary(base, control_wires=[0, 1])
+
+    Typically controlled operations apply a desired gate if the control qubits
+    are all in the state :math:`\vert 1\rangle`. However, there are some situations where
+    it is necessary to apply a gate conditioned on all qubits being in the
+    :math:`\vert 0\rangle` state, or a mix of the two.
+
+    The state on which to control can be changed by passing a string of bits to
+    `control_values`. For example, if we want to apply a single-qubit unitary to
+    wire ``3`` conditioned on three wires where the first is in state ``0``, the
+    second is in state ``1``, and the third in state ``1``, we can write:
+
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1, 2], wires=3, control_values=[0, 1, 1])
+
+    or
+
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1, 2], wires=3, control_values=[False, True, True])
+    """
+    num_wires = AnyWires
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (2,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    # pylint: disable= too-many-arguments
+    def __init__(
+        self,
+        base,
+        control_wires=None,
+        wires=None,
+        control_values=None,
+        do_queue=True,
+        unitary_check=False,
+        work_wires=None,
```

Comment:
Not used

Reply:
Could just make sure it gets passed to `super().__init__`.  While we don't use work wires much right now outside of `MultiControlledX`, decomposition algorithms will need it when we start having them.

Reply:
Thanks, yes, it's meant to be passed to the `super().__init__` function. Updated now.